### PR TITLE
Gift card fixes

### PR DESCRIPTION
**Pull request by hf2186**
Minor fixes and updates

### CODE REVIEW

File: app/models/spree/order_decorator.rb [hunk 1]
```diff
@@ -12,29 +14,8 @@ def gift_card_match(line_item, options)
             gc_detail_set.superset?(options_set)
           end
         end
-
```
Comment (jaredmurphy) [inline app/models/spree/order_decorator.rb:15]: why are we removing the `finalize!` method from order?

Comment (hf2186) [reply app/models/spree/order_decorator.rb:15]: because we should not override this and use the state machine to prevent the race condition

Comment (jaredmurphy) [reply app/models/spree/order_decorator.rb:15]: 👍

**Review by jaredmurphy — APPROVED** [review]

**Pull request merged**